app/src/features/apiClient/screens/apiClient/components/sidebar/components/collectionsList/CollectionsList.tsx

```diff
@@ -335,6 +342,59 @@ export const CollectionsList: React.FC<Props> = ({ onNewClick, recordTypeToBeCre
     );
   }, [collectionId, requestId, apiClientRecords]);
 
+  const handleDropToRoot = useCallback(
+    async (item: DraggableApiRecord) => {
+      try {
+        await moveRecords({
+          collectionId: "",
+          recordsToMove: [item.record],
+        });
+
+        toast.success(`Moved "${item.record.name}" to workspace root`);
+      } catch (error) {
+        toast.error((error as any)?.message || "Failed to move item. Please try again.");
+      }
+    },
+    [moveRecords]
+  );
+
+  // Shared drop configuration for all drop zones
+  const dropConfig = {
```

Comment:
If component re-renders then this object will be newly created, wrap it with `useMemo`